File: tests/algorithms/x/PHP/data_structures/hashing/number_theory/prime_numbers.php
<?php
ini_set('memory_limit', '-1');
function isPrime($number) {
  if ($number < 2) {
  return false;
}
  if ($number < 4) {
  return true;
}
  if ($number % 2 == 0) {
  return false;
}
  $i = 3;
  while ($i * $i <= $number) {
  if ($number % $i == 0) {
  return false;
}
  $i = $i + 2;
};
  return true;
}
function nextPrime($value, $factor, $desc) {
  $v = $value * $factor;
  $firstValue = $v;
  while (!isPrime($v)) {
  if ($desc) {
  $v = $v - 1;
} else {
  $v = $v + 1;
}
};
  if ($v == $firstValue) {
  if ($desc) {
  return nextPrime($v - 1, 1, $desc);
} else {
  return nextPrime($v + 1, 1, $desc);
};
}
  return $v;
}
echo rtrim(json_encode(isPrime(0), 1344)), PHP_EOL;
echo rtrim(json_encode(isPrime(1), 1344)), PHP_EOL;
echo rtrim(json_encode(isPrime(2), 1344)), PHP_EOL;
echo rtrim(json_encode(isPrime(3), 1344)), PHP_EOL;
echo rtrim(json_encode(isPrime(27), 1344)), PHP_EOL;
echo rtrim(json_encode(isPrime(87), 1344)), PHP_EOL;
echo rtrim(json_encode(isPrime(563), 1344)), PHP_EOL;
echo rtrim(json_encode(isPrime(2999), 1344)), PHP_EOL;
echo rtrim(json_encode(isPrime(67483), 1344)), PHP_EOL;
echo rtrim(json_encode(nextPrime(14, 1, false), 1344)), PHP_EOL;
echo rtrim(json_encode(nextPrime(14, 1, true), 1344)), PHP_EOL;

File: tests/algorithms/x/Java/data_structures/hashing/number_theory/prime_numbers.java
public class Main {

    static boolean isPrime(int number) {
        if (number < 2) {
            return false;
        }
        if (number < 4) {
            return true;
        }
        if (Math.floorMod(number, 2) == 0) {
            return false;
        }
        int i = 3;
        while (i * i <= number) {
            if (Math.floorMod(number, i) == 0) {
                return false;
            }
            i = i + 2;
        }
        return true;
    }

    static int nextPrime(int value, int factor, boolean desc) {
        int v = value * factor;
        int firstValue = v;
        while (!(Boolean)isPrime(v)) {
            if (((Boolean)(desc))) {
                v = v - 1;
            } else {
                v = v + 1;
            }
        }
        if (v == firstValue) {
            if (((Boolean)(desc))) {
                return nextPrime(v - 1, 1, desc);
            } else {
                return nextPrime(v + 1, 1, desc);
            }
        }
        return v;
    }
    public static void main(String[] args) {
        System.out.println(isPrime(0));
        System.out.println(isPrime(1));
        System.out.println(isPrime(2));
        System.out.println(isPrime(3));
        System.out.println(isPrime(27));
        System.out.println(isPrime(87));
        System.out.println(isPrime(563));
        System.out.println(isPrime(2999));
        System.out.println(isPrime(67483));
        System.out.println(nextPrime(14, 1, false));
        System.out.println(nextPrime(14, 1, true));
    }
}
